user/id.go

```diff
@@ -0,0 +1,118 @@
+package user
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/mr-tron/base58"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/hash"
+	"github.com/nspcc-dev/neo-go/pkg/encoding/address"
+	"github.com/nspcc-dev/neofs-api-go/v2/refs"
+)
+
+// ID identifies users of the NeoFS system.
+//
+// ID is mutually compatible with github.com/nspcc-dev/neofs-api-go/v2/refs.OwnerID
+// message. See ReadFromV2 / WriteToV2 methods.
+//
+// Instances can be created using built-in var declaration. Zero ID is not valid,
+// so it MUST be initialized using some modifying function (e.g. Decode method,
+// EncodePublicKeyToID function, etc.).
+//
+// Note that direct typecast is not safe and may result in loss of compatibility:
+// 	_ = ID([25]byte{}) // not recommended
+type ID [25]byte
+
+// ReadFromV2 reads ID from the refs.OwnerID message. Returns an error if
+// the message is malformed according to the NeoFS API V2 protocol.
+//
+// See also WriteToV2.
+func (x *ID) ReadFromV2(m refs.OwnerID) error {
+	return x.Decode(m.GetValue())
+}
+
+// WriteToV2 writes ID to the refs.OwnerID message.
+// The message must not be nil.
+//
+// See also ReadFromV2.
+func (x ID) WriteToV2(m *refs.OwnerID) {
+	m.SetValue(x[:])
+}
+
+// Encode encodes ID into binary Neo3 wallet address.
+//
+// Uses first 25 bytes of dst, so its length MUST NOT be less than 25.
```

Comment:
how about [panic](https://github.com/nspcc-dev/neofs-sdk-go/pull/202#discussion_r848707291)?

Reply:
Implementation panics. `MUST NOT` implies potential panic, do u think we need to doc this case clearly?

Reply:
> do u think we need to doc this case clearly?

the idea was to keep in sync comments. if `oid`, `cid` are documented with "panic" I suggest adding that info here. if the `used` documents it as "must not" that `oid` and `cid` should be commented in the same way

Reply:
All these types shouldn't follow the same design pattern, they just similar in current implementation.